src/extractor/guidance/turn_handler.cpp

```diff
@@ -227,26 +215,6 @@ Intersection TurnHandler::handleComplexTurn(const EdgeID via_edge, Intersection
                                      via_edge,
                                      isThroughStreet(obvious_index, intersection),
                                      intersection[obvious_index]);
-        if (has_same_name_turn &&
```

Comment:
The obvious turn discovery, when first written, did not consider roads that continue as well as they do now. This handling here was more of a hack to make cases work where the road continues to a side but the most obvious turn would be a straight turn.

With the changes done to `findObviousTurn()`, this hack breaks `features/guidance/turn.feature:815`. [here](https://github.com/Project-OSRM/osrm-backend/blob/master/features/guidance/turn.feature#L815-L841), turning https://github.com/Project-OSRM/osrm-backend/blob/master/features/guidance/turn.feature#L840 into `depart,turn straight,arrive`. The interaction requires the removal of the hack (which is an added benefit in my view)